repose-aggregator/components/filters/keystone-v2-filter/src/main/scala/org/openrepose/filters/keystonev2/KeystoneV2Authorization.scala

```diff
@@ -59,19 +56,30 @@ object KeystoneV2Authorization extends LazyLogging {
     }
   }
 
-  val handleFailures: PartialFunction[Try[Unit.type], KeystoneV2Result] = {
-    case Failure(e: InvalidTenantException) => Reject(SC_UNAUTHORIZED, Some(e.getMessage))
-    case Failure(e: UnauthorizedEndpointException) => Reject(SC_FORBIDDEN, Some(e.getMessage))
-    case Failure(e: UnparseableTenantException) => Reject(SC_UNAUTHORIZED, Some(e.getMessage))
+  // TODO: Switch to getRequestTenants (plural) and include all tenants extracted. This is a behavior change.
+  def getRequestTenant(config: ValidateTenantType, request: HttpServletRequestWrapper): String = {
+    Option(config).flatMap(validateTenantConfig =>
+      Option(validateTenantConfig.getHeaderExtractionName).flatMap(headerName =>
+        request.getSplittableHeaderScala(headerName).headOption
```

Comment:
Pretty sure these are still going to have qualities dangling off them.

Reply:
Dang it. Totally forgot about that. Will fix.